### PR TITLE
Use WebDriver instead of PhpBrowser to allow usage of other Drivers

### PR DESCRIPTION
**Pull request by maciejmrozinski**

### CODE REVIEW

**Comment by Naktibalda** [comment]
Please explain what are you trying to do here.

**Comment by maciejmrozinski** [comment]
I want to use Facebook module, but with Chrome WebDriver and not PHPBrowser. When i declare my config as using WebDriver and that Facebook module depends on WebDriver i get an error that PHPBrowser was expected.

Facebook module uses common methods from WebDriver module, so no need to strict it to PHPBrowser only.

**Comment by Naktibalda** [comment]
But your change restricts Facebook module to WebDriver, so it no longer works with PhpBrowser.

Please make it work with both.

File: src/Codeception/Module/Facebook.php [hunk 1]
```diff
@@ -121,10 +121,10 @@ public function _requires()
 
     public function _depends()
     {
-        return ['Codeception\Module\PhpBrowser' => $this->dependencyMessage];
+        return ['Codeception\Module\WebDriver' => $this->dependencyMessage];
```
Comment (DavertMik) [inline src/Codeception/Module/Facebook.php]: Please use `Codeception\Lib\Interfaces\Web`, as it will match both PhpBrowser and WebDriver

**Comment by maciejmrozinski** [comment]
To cover all used methods, `Codeception\Lib\Interfaces\Web` and `Codeception\Lib\Interfaces\Remote` need to be checked. What do You think?

File: src/Codeception/Module/Facebook.php [hunk 2]
```diff
@@ -121,11 +123,17 @@ public function _requires()
 
     public function _depends()
     {
-        return ['Codeception\Module\PhpBrowser' => $this->dependencyMessage];
+        return ['Codeception\Lib\Interfaces\Remote|Codeception\Lib\Interfaces\Remote' => $this->dependencyMessage];
```
Comment (DavertMik) [inline src/Codeception/Module/Facebook.php]: I think it won't work this way :(
Our parser is not so smart to handle both interfaces... Especially two the same interface...
Let's ask for `Codeception\Lib\Interfaces\Web` interface here

File: src/Codeception/Module/Facebook.php [hunk 3]
```diff
@@ -121,11 +123,17 @@ public function _requires()
...
     }
 
-    public function _inject(PhpBrowser $browserModule)
+    public function _inject($browserModule)
     {
+        if (!$browserModule instanceof Remote || !$browserModule instanceof Web) {
```
Comment (DavertMik) [inline src/Codeception/Module/Facebook.php]: And check if passed a browserModule is an instance of `Remote`, if not throw extension as you do it here

**Comment by Naktibalda** [comment]
Any updates?

**Pull request closed without merging**